x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -19,22 +21,51 @@ export const resolverMiddlewareFactory: MiddlewareFactory = context => {
   return api => next => async (action: ResolverAction) => {
     next(action);
     if (action.type === 'userChangedSelectedEvent') {
-      if (context?.services.http) {
+      if (context?.services.http && action.payload.selectedEvent) {
         api.dispatch({ type: 'appRequestedResolverData' });
-        const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
-        const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
-        const [{ lifecycle }, { children }, { events: relatedEvents }] = await Promise.all([
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
-            query: { legacyEndpointID },
-          }),
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`, {
-            query: { legacyEndpointID },
-          }),
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`, {
-            query: { legacyEndpointID },
-          }),
-        ]);
-        const response = [...lifecycle, ...children, ...relatedEvents];
+        let response = [];
+        let lifecycle: ResolverEvent[];
+        let childEvents: ResolverEvent[];
+        let relatedEvents: ResolverEvent[];
+        let children = [];
+        const ancestors: ResolverEvent[] = [];
+        const maxAncestors = 5;
+        if (isLegacyEvent(action.payload.selectedEvent)) {
+          const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
+          const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
+          [{ lifecycle }, { children }, { events: relatedEvents }] = await Promise.all([
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
+              query: { legacyEndpointID },
+            }),
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`, {
+              query: { legacyEndpointID },
+            }),
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`, {
+              query: { legacyEndpointID },
+            }),
+          ]);
+          childEvents = children.length > 0 ? children.map((child: any) => child.lifecycle) : [];
+        } else {
+          const uniquePid = action.payload.selectedEvent.endpoint.process.entity_id;
+          const ppid = action.payload.selectedEvent.endpoint.process.parent?.entity_id;
+          async function getAncestors(pid: string | undefined) {
```

Comment:
It might not be worth changing since I think we're going to hook this up to Andrew's backend anyway which will require changes to this function but the backend allows an `ancestors` query param that can effectively do what this function is doing I think: https://github.com/elastic/kibana/blob/master/x-pack/plugins/endpoint/server/routes/resolver/lifecycle.ts#L39

Just pass in `ancestors=5` and it'll walk up the tree as far as it can up to 5.

Reply:
Ah that is much better, changing to do that